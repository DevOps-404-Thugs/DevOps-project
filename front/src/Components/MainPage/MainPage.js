import React, { useEffect, useState } from 'react'
import Axios from 'axios'
import {Col, Row } from 'antd';
import { Link } from 'react-router-dom';
// import NavBar from '../NavBar/NavBar.js';
import "./MainPage.css"


function MainPage() {

    const [Houses,setHouses] = useState([])//set as array 

    
		useEffect(() => {
			Axios.get(`http://127.0.0.1:8000/housings`)
				.then(response => {
          console.log(response.data);
          console.log("housingsGET:"+response.status);
					setHouses(response.data);
				})
		},[]);
		


    const renderCards = Houses.map((house, index) => {

        return <Col lg={6} md={8} xs={24} >
                  <img style={{ width:'50%'}} src={`http://bootstrapmade.com/demo/themes/EstateAgency/assets/img/slide-2.jpg`} alt={"houseImage"}/>
                  <p>{house.name}</p>
                  <p>{house.address}</p>

                  {/* <Link to={'/detail/'+house._id}>Go to see detail</Link> */}
                  <Link to={{
                    pathname: '/detail',
                    state: {
                      objectId: house._id.$oid
                    }
                  }}>Go to see detail</Link>

        </Col>
    })


    return (
        <div style={{ width: '75%', margin: '3rem auto' }}>
          {/* <NavBar /> */}
            <div style={{ textAlign: 'center' }}>
                <h2>  Find Your Future Home in iHomie </h2>
            </div>

            <div>
              <Row gutter={[16, 16]}>
                {renderCards}
              </Row>
            </div>
            <br /><br />

            <div style={{ display: 'flex', justifyContent: 'center' }}>
                {/* <button onclick={() => this.props.history.push('upload')}>Upload New House</button> */}
                <Link className="link" to="/upload">Upload New House</Link>
            </div>

        </div>
    )
}

export default MainPage;