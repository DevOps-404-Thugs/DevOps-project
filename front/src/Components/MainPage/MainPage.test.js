import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MainPage from './MainPage';


describe('test', () =>{
    test("match the snapshot", () =>{
        const component = renderer.create(
            <BrowserRouter>
                <MainPage/>
            </BrowserRouter>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });  
});