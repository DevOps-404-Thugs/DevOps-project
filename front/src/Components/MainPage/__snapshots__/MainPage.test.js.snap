// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mainpage component match the snapshot 1`] = `
<div
  style={
    Object {
      "margin": "3rem auto",
      "width": "65%",
    }
  }
>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <h2>
        Find Your Future Home in iHomie 
    </h2>
  </div>
  <div>
    <div
      className="ant-row"
      style={
        Object {
          "marginBottom": 8,
          "marginLeft": -8,
          "marginRight": -8,
          "marginTop": -8,
        }
      }
    />
  </div>
  <br />
  <br />
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <a
      className="link"
      href="/upload"
      onClick={[Function]}
    >
      Upload New House
    </a>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <a
      className="link"
      href="/account_update"
      onClick={[Function]}
      style={
        Object {
          "color": "orange",
        }
      }
    >
      Update Account Information
    </a>
  </div>
</div>
`;
